main.go

```diff
@@ -0,0 +1,304 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+)
+
+const (
+	staffCategory  = 29
+	buildIssuesCat = 11
+	stateFile      = "data.txt"
+	discourseDone  = 1
+	discourseLog   = "      Migrated to Discourse: %s"
+	commentDone    = 2
+	commentLog     = "      Commented on issue"
+	closeDone      = 3
+	closeLog       = "      Closed GitHub issue"
+	lockDone       = 4
+	lockLog        = "      Locked GitHub issue"
+	maxCount       = 1
+	activeTpl      = "Hi %s! We are migrating our GitHub issues to Discourse (https://discuss.bitrise.io/c/issues/build-issues). From now on, you can track this issue at: %s"
+	staleTpl       = "Hi %s! We are migrating our GitHub issues to Discourse (https://discuss.bitrise.io/c/issues/build-issues). Because this issue has been inactive for more than three months, we will be closing it. If you feel it is still relevant, please open a ticket on Discourse!"
+)
+
+type RestoredIssue struct {
+	URL    string
+	Owner  string
+	Repo   string
+	IssNum int
+	Done   int
+	Extra  string
+}
+
+type Repo struct {
+	Owner string
+	Name  string
+}
+
+type Step struct {
+	LatestVersionNumber string `json:"latest_version_number"`
+	Versions            map[string]map[string]interface{}
+}
+
+type Spec struct {
+	Steps map[string]Step
+}
+
+func prefixWithRunID(str string) string {
+	return fmt.Sprintf("[TEST][%s] %s", time.Now().Format(time.RFC3339), str)
+}
+
+var c, staleCount, activeCount, cPR = 0, 0, 0, 0
```

Comment:
i am not that happy to see these global variables, some of them are not even used in this file. Please move as many of these as you can and as close to the usage as possible